torchtitan/components/optimizer.py

```diff
@@ -175,8 +182,72 @@ def zero_grad(self) -> None:
         pass
 
 
+class FTOptimizersContainer(OptimizersContainer):
+    def __init__(
+        self,
+        model_parts: List[nn.Module],
+        optimizer_cls: type[T],
+        optimizer_kwargs: Dict[str, Any],
+        ft_manager: Optional["ft.Manager"],
+    ) -> None:
+        super().__init__(model_parts, optimizer_cls, optimizer_kwargs)
+
+        # Force to initialize the optimizer state so that `optim.step()`
+        # won't be called by state_dict() and load_state_dict().
+        _ = {
+            k: v
+            for sd in map(get_optimizer_state_dict, model_parts, self.optimizers)
+            for k, v in sd.items()
+        }
+        self.cache_state_dict: Dict[str, Any] = {}
+        self._ft_optimizer = ft.Optimizer(ft_manager, self)
```

Comment:
This might be janky as hell -- but you can pass `super()` as a variable instead of `self` hand have ft.Optimizer call the parent classes directly

Reply:
ye, that's another way I was considering. Not sure which one is more explicit for people.